File: Hydrogene/All/SharedUI/SharedUI.Shared/MainWindowController.WPF.cs
namespace SharedUI.Shared
{
	#if ECHOES
	using System.Windows;
	using System.Windows.Controls;

	public partial class MainWindowController
	{
		public this()
		{
			window = new MainWindow withController(this);
			setup();
		}

		public Window window { get; }

		//
		// Compatibility Helpers. These could/should be in a shared base class, in a real app with many window/views
		//

		public void showWindow(id sender)
		{
			window.Show();
		}

		//
		// Add WPF-specific code here
		//
	}
	#endif
}

File: Oxygene/All/SharedUI/SharedUI.Shared/MainWindowController.WPF.pas
namespace SharedUI.Shared;
{$IF ECHOES}

uses
  System.Windows,
  System.Windows.Controls;

type
  MainWindowController = public partial class
  public

    property window: Window; readonly;

    constructor;
    begin
      window := new MainWindow withController(self);
      setup();
    end;

    //
    // Compatibility Helpers. These could/should be in a shared base class, in a real app with many window/views
    //

    method showWindow(sender: id);
    begin
      window.Show();
    end;

    //
    // Add WPF-specific code here
    //

  end;

{$ENDIF}
end.